src/components/app/details/AboutTagEditModal.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useRef, useState } from 'react'
+import { ReactComponent as Close } from '../../../assets/icons/ic-cross.svg'
+import { Drawer, Progressing, showError } from '../../common'
+import TagLabelSelect from './TagLabelSelect'
+import { AboutAppInfoModalProps, TagType } from '../types'
+import { createAppLabels } from '../service'
+import { toast } from 'react-toastify'
+
+export default function AboutTagEditModal({
+    isLoading,
+    appId,
+    onClose,
+    currentLabelTags,
+    getAppMetaInfoRes,
+}: AboutAppInfoModalProps) {
+    const editLabelRef = useRef(null)
+    const [submitting, setSubmitting] = useState(false)
+    const [labelTags, setLabelTags] = useState<TagType[]>(currentLabelTags || [])
+
+    const handleSaveAction = async (e): Promise<void> => {
+        e.preventDefault()
+        const _labelTags = []
+        let invalidLabels = false
+        for (let index = 0; index < labelTags.length; index++) {
+            const element = labelTags[index]
+            if (element.isInvalidKey || element.isInvalidValue) {
+                invalidLabels = true
+                break
+            } else if (element.key) {
+                _labelTags.push({ key: element.key, value: element.value, propagate: element.propagate })
+            }
+        }
```

Comment:
**Suggestion:** This can be moved to a separate function & that function can return {_labelTags, invalidLabels}

Reply:
will do it in next PR